File: Python/Easy/Filter/Array/136_Single_Number.py
def singleNumber(nums: list[int]) -> int:
    count = {}
    for num in nums:
        count[num] = count.get(num, 0) + 1
    for k, v in count.items():
        if v == 1:
            return k



nums = [2, 2, 1]
print(singleNumber(nums))

File: PHP/Easy/Filters/Bit_Manipulation/136-Single_Number.php
<?php
//function singleNumber($nums)
//{
//    $result = 0;
//    foreach ($nums as $num) {
//        $result ^= $num;
//    }
//    return $result;
//}
function singleNumber($nums)
{
    $count = [];
    foreach ($nums as $num) {
        if (array_key_exists($num, $count)) {
            $count[$num] += 1;
        } else {
            $count[$num] = 1;
        }
    }
    foreach ($count as $num => $value) {
        if ($value = 1) {
            return $num;
        }
    }
}

$nums = [4, 1, 2, 1, 2];
echo "\n";
print_r(singleNumber($nums));
echo "\n";